cdap-unit-test/src/test/java/co/cask/cdap/test/app/TestFrameworkTestRun.java

```diff
@@ -1240,18 +1240,24 @@ public void testAppWithTxTimeout() throws Exception {
     int txDefaulTimeoutMapReduce = 23;
     int txDefaulTimeoutSpark = 24;
 
-    ApplicationManager appManager = deployApplication(testSpace, AppWithCustomTx.class);
+    final ApplicationManager appManager = deployApplication(testSpace, AppWithCustomTx.class);
     try {
       // attempt to start with a tx timeout that exceeds the max tx timeout
       try {
         appManager.getServiceManager(AppWithCustomTx.SERVICE)
           .start(txTimeoutArguments(100000));
       } catch (IllegalArgumentException e) {
-        //expected
+        //expected, this will finally cause AppWithCustomTx.SERVICE to have FAILED status
       }
-
       // now start all the programs with valid tx timeouts
       getStreamManager(testSpace.stream(AppWithCustomTx.INPUT)).send("hello");
+      // wait for the failed status of AppWithCustomTx.SERVICE to be persisted, so that it can be started again
+      Tasks.waitFor(1, new Callable<Integer>() {
```

Comment:
This change is fine, but it seems like this 1-liner would've been simpler:
`serviceManager.waitForRun(ProgramRunStatus.FAILED, 30, TimeUnit.SECONDS);`